### PR TITLE
Add support for clips4sale

### PR DESCRIPTION
**Pull request by IAmKontrast**
Added studio and scene support for `clips4sale`.

**Error on icon-list-view repeated reloading:**
Unfortunately I wasn't able to solve the `Minified React error` `#418`, `#423` and `#425` on the icon-list-view on repeated reloading.
To reproduce go on one of the studio profile pages (/studio/) and then switch to the icon-list-view. Everything works fine and all checks for scenes went on. Reloading the page causes the errors.
Will be triggered by this [`check(Target.Scene, "div[class*='w-full'] a[class*='search-clip__title']", { observe: true });`](https://github.com/IAmKontrast/stash-checker/blob/a299c907614281c217ea876060d9dde4199ed7cd/src/stashChecker.ts#L373
)

### CODE REVIEW

File: src/stashChecker.ts [hunk 1]
```diff
@@ -363,6 +363,16 @@ export async function runStashChecker() {
             check(Target.Scene, "[class*='pcVideoListItem'] span.title a[href*='/view_video.php?viewkey=']", { observe: true })
             break;
         }
+        case "www.clips4sale.com": {
+            check(Target.Studio, "div[class*='items-center'] h1", { urlSelector: currentSite });
+            check(Target.Studio, "div[class*='items-center'] div[class*='flex']:not([class*='hidden']) a[href*='/studio/']", {
```
Comment (timo95) [inline src/stashChecker.ts:368]: This seems to match the title as well, resulting in additional queries that will be overwritten by studio queries. Maybe use `[data-testid*='studioAnchor'] a[href*='/studio/']` instead.

File: src/stashChecker.ts [hunk 2]
```diff
@@ -363,6 +363,16 @@ export async function runStashChecker() {
...
+                observe: true,
+                urlSelector: e => closestUrl(e)?.split("/Cat")?.[0]
+            });
+            check(Target.Scene, "figure[class*='mediabook-preview'] h1[class*='static group']", { urlSelector: currentSite });
+            check(Target.Scene, "div[class*='w-full'] a[class*='search-clip__title']", { observe: true });
```
Comment (timo95) [inline src/stashChecker.ts:373]: On the recently added clips page, when you scroll down below the side menu, nothing is matched. `div.w-full a.search-clip__title, div.static a.search-clip__title` would include everything.

**Review by timo95 — CHANGES_REQUESTED** [review]
I tried it without your changes and `Minified React error` still occurs, just much less often. Annoying, but not really our problem.

Please fix the overlapping Studio/Scene checks. I am using this addon to test css selectors, maybe that helps: https://addons.mozilla.org/de/firefox/addon/try-xpath/

**Comment by IAmKontrast** [comment]
Completely restart my approach with element selectors from classes to `data-testid` attributes. Now there should be any additional or overriding queries and also a performance improvement is noticeable.
Maybe they fixed something on the website because the Minified React errors are gone.

> On the recently added clips page, when you scroll down below the side menu, nothing is matched.

This works now as well as switch between tile and list view and sorting.

**Comment by timo95** [comment]
> a performance improvement is noticeable

Thats just the merge from master

**Comment by IAmKontrast** [comment]
Add a fix for studio/category matching with #46

**Pull request merged**